src/lib.rs

```diff
@@ -89,6 +89,68 @@ pub mod unix {
     pub use crate::sys::SourceFd;
 }
 
+cfg_any_os_util! {
+/// Windows-only extensions to the mio crate.
+///
+/// Mio on windows is currently implemented with IOCP for a high-performance
+/// implementation of asynchronous I/O. On top of the IOCP, a connection to the "Ancillary Function
+/// Driver" provides socket services. However, this approach doesn't work for other handle types
+/// like files and named pipes. They need to be connected directly to the IO completion port like
+/// this is done on unix-like systems  with the `SourceFd` type. The purpose of this module is to
+/// similarly provide a mechanism for foreign I/O types to get hooked up into the IOCP event loop.
+///
+/// This module provides types for interfacing with a custom IOCP handle:
+///
+/// * `Binding` - this type is intended to govern binding with mio's `Registry`
+///   type. Each I/O object should contain an instance of `Binding` that's
+///   interfaced with for the implementation of the `Source` trait. The
+///   `register`, `reregister`, and `deregister` methods for the `Source` trait
```

Comment:
I see that `Binding` only has `register_handle`, how would I implement de/re-register?

Or this is that part simply not implemented yet?

Reply:
Its not implemented yet because I'm unsure how that should/could be done as in Windows, there is no way of removing a handle from an IOCP except by closing that handle. So the only thing we should do here is to remove the handler from the list of IO handlers, which should be done by a Drop implementation of that type. reregistration would only support changing the token; I will add both functionalities.